lib/improver/nowcasting/utilities.py

```diff
@@ -0,0 +1,214 @@
+
+        Raises:
+            ValueError: Operation not supported.
+
+        """
+        # A minimum precipitation rate in mm/h that will be used as a lower
+        # precipitation rate threshold.
+        self.min_precip_rate_mmh = 1/32.
+
+        possible_operations = ['+', 'add', '-', 'subtract']
+
+        if operation in possible_operations:
+            self.operation = operation
+        else:
+            msg = ("Operation '{}' not supported for combining "
+                   "precipitation rate and "
+                   "orographic enhancement.".format(operation))
+            raise ValueError(msg)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<ApplyOrographicEnhancement: operation: {}>')
+        return result.format(self.operation)
+
+    @staticmethod
+    def _select_orographic_enhancement_cube(precip_cube, oe_cubes):
+        """Select the orographic enhancement cube with the required time
+        coordinate.
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Cube containing the input precipitation fields.
+            oe_cubes (iris.cube.Cube or iris.cube.CubeList):
+                Cube or CubeList containing the orographic enhancement fields.
+
+        Returns:
+            oe_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement fields at the
+                required time.
+
+        """
+        time_point, = iris_time_to_datetime(precip_cube.coord("time").copy())
+        oe_cube = extract_nearest_time_point(oe_cubes, time_point)
+        return oe_cube
+
+    def _apply_orographic_enhancement(self, precip_cube, oe_cube):
+        """Combine the precipitation rate cube and the orographic enhancement
+        cube.
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Cube containing the input precipitation field.
+            oe_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement field matching
+                the validity time of the precipitation cube.
+
+        Returns:
+            cube (iris.cube.Cube):
+                Cube containing the precipitation rate field modified by the
+                orographic enhancement cube.
+
+        """
+        # Ensure the orographic enhancement cube matches the
+        # dimensions of the precip_cube.
+        oe_cube = check_cube_coordinates(precip_cube, oe_cube)
+
+        # Ensure that orographic enhancement is in the units of the
+        # precipitation rate cube.
+        oe_cube.convert_units(precip_cube.units)
+
+        # Set orographic enhancement to be zero for points with a
+        # precipitation rate of < 1/32 mm/hr.
+        original_units = Unit("mm/hr")
+        threshold_in_cube_units = (
+            original_units.convert(self.min_precip_rate_mmh,
+                                   precip_cube.units))
+        oe_cube.data[precip_cube.data < threshold_in_cube_units] = 0.
```

Comment:
I don't think we should be changing the orographic enhancement data. If we apply the same orographic enhancement cubelist to multiple precip_cubes in the same process, this will not work as intended. I think it would be better to either take a copy of the oe_cube unless you can think of a more efficient way (something like the mask method in the following function?).

Reply:
I've added a copy of the orographic enhancement cube. I checked the behaviour in this method and although the orographic enhancement had not been set to zero after this method had completed, the unit conversion a few lines higher within this method was occurring in-place.